src/main/java/org/mule/service/scheduler/internal/DefaultSchedulerService.java

```diff
@@ -89,116 +92,106 @@ public String getName() {
 
   @Override
   public Scheduler cpuLightScheduler() {
-    checkStarted();
-    final SchedulerConfig config = config();
-    pollsReadLock.lock();
-    try {
-      return poolsByConfig.get(getInstance())
-          .createCpuLightScheduler(config, cpuBoundWorkers(), resolveStopTimeout(config));
-    } finally {
-      pollsReadLock.unlock();
-    }
+    return cpuLightScheduler(config(), getInstance(), null, null);
   }
 
   @Override
-  public Scheduler ioScheduler() {
-    checkStarted();
-    final SchedulerConfig config = config();
-    pollsReadLock.lock();
-    try {
-      return poolsByConfig.get(getInstance())
-          .createIoScheduler(config, ioBoundWorkers(), resolveStopTimeout(config));
-    } finally {
-      pollsReadLock.unlock();
-    }
+  public Scheduler cpuLightScheduler(SchedulerConfig config) {
+    return cpuLightScheduler(config, getInstance(), null, null);
   }
 
   @Override
-  public Scheduler cpuIntensiveScheduler() {
+  public Scheduler cpuLightScheduler(@Named(OBJECT_SCHEDULER_BASE_CONFIG) SchedulerConfig config,
+                                     SchedulerPoolsConfigFactory poolsConfigFactory) {
+    return cpuLightScheduler(config, poolsConfigFactory, null, null);
+  }
+
+  @Inject
+  public Scheduler cpuLightScheduler(@Named(OBJECT_SCHEDULER_BASE_CONFIG) SchedulerConfig config,
+                                     SchedulerPoolsConfigFactory poolsConfigFactory,
+                                     FeatureFlaggingService featureFlaggingService, ProfilingService profilingService) {
     checkStarted();
-    final SchedulerConfig config = config();
     pollsReadLock.lock();
     try {
-      return poolsByConfig.get(getInstance())
-          .createCpuIntensiveScheduler(config, cpuBoundWorkers(), resolveStopTimeout(config));
+      return poolsByConfig.get(poolsConfigFactory)
+          .createCpuLightScheduler(config, cpuBoundWorkers(), resolveStopTimeout(config),
+                                   getProfilingService(featureFlaggingService, profilingService));
     } finally {
       pollsReadLock.unlock();
     }
   }
 
-  @Override
-  public Scheduler cpuLightScheduler(SchedulerConfig config) {
-    checkStarted();
-    pollsReadLock.lock();
-    try {
-      return poolsByConfig.get(getInstance())
-          .createCpuLightScheduler(config, cpuBoundWorkers(), resolveStopTimeout(config));
-    } finally {
-      pollsReadLock.unlock();
+  /**
+   * Returns the {@link ProfilingService} only if the profiling is enabled.
+   * 
+   * @param featureFlaggingService The {@link FeatureFlaggingService}
+   * @param profilingService       The {@link ProfilingService}
+   * @return The given {@link ProfilingService} or null if the profiling feature is disabled.
+   */
+  private ProfilingService getProfilingService(FeatureFlaggingService featureFlaggingService, ProfilingService profilingService) {
```

Comment:
why not encapsulate this logic in the runtime, when determining the profilingService to pass?